### DISCUSSION TITLE
LAB 2.01 TASK 4

### DISCUSSION THREAD

**Discussion by gitogel** [rramosp/ai4eng.v1#251] [02 - PYTHON]
Hola Profesor:¿Podría ayudarme a entender qué está mal acá?. Mi código aparentemente funciona bien, sin embargo la plataforma me pone este mensaje: 
task_04 submitted. your grade is 0
----- grader message -------
incorrect: testing with 100 random calls
incorrect answer. input was 6 but you returned True, correct answer was True
![CÓDIGO](https://user-images.githubusercontent.com/126641009/230182867-9fb15fdf-e220-41a5-a5ca-f928e6681c31.jpg)

**Comment by rramosp** [comment]
tienes que devolver True, como booleano, no como string

On Wed, Apr 5, 2023 at 2:16 PM gitogel ***@***.***> wrote:

> Hola Profesor:¿Podría ayudarme a entender qué está mal acá?. Mi código
> aparentemente funciona bien, sin embargo la plataforma me pone este mensaje:
> task_04 submitted. your grade is 0
> ----- grader message -------
> incorrect: testing with 100 random calls
> incorrect answer. input was 6 but you returned True, correct answer was
> True
>
> Este es mi código:
> def perfecto(n):
> #tu codigo aqui, pista:
> # halle todos los divisores de n y sumelos en una variable auxiliar,
> #compare este resultado con el valor original
> l=list(range(1,n-1))
> i=1
> t=0
> for i in range(len(l)):
> if n%l[i]==0:
> t=t+l[i]
> i+=1
> if t==n:
> return("True")
> else:
> return("False")
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/251>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADJRZIJLXCFRUIQUGIDRKKTW7XAHBANCNFSM6AAAAAAWUPBAPY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

-- 
*Aviso legal:* El contenido de este mensaje y los archivos adjuntos son 
confidenciales y de uso exclusivo de la Universidad Nacional de Colombia. 
Se encuentran dirigidos sólo para el uso del destinatario al cual van 
enviados. La reproducción, lectura y/o copia se encuentran prohibidas a 
cualquier persona diferente a este y puede ser ilegal. Si usted lo ha 
recibido por error, infórmenos y elimínelo de su correo. Los Datos 
Personales serán tratados conforme a la Ley 1581 de 2012 y a nuestra 
Política de Datos Personales que podrá consultar en la página web 
www.unal.edu.co <http://www.unal.edu.co/>.* *Las opiniones, informaciones, 
conclusiones y cualquier otro tipo de dato contenido en este correo 
electrónico, no relacionados con la actividad de la Universidad Nacional de 
Colombia, se entenderá como personales y de ninguna manera son avaladas por 
la Universidad.
